test/functional/client_side_encryption_prose_tests.js

```diff
@@ -0,0 +1,974 @@
+const chai = require('chai');
+const expect = chai.expect;
+chai.use(require('chai-subset'));
+
+// Tests for the ClientEncryption type are not included as part of the YAML tests.
+
+// In the prose tests LOCAL_MASTERKEY refers to the following base64:
+
+// .. code:: javascript
+
+//   Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk
+
+describe(
+  'Client Side Encryption Prose Tests',
+  { requires: { clientSideEncryption: true } },
+  function() {
+    const dataDbName = 'db';
+    const dataCollName = 'coll';
+    const dataNamespace = `${dataDbName}.${dataCollName}`;
+    const keyVaultDbName = 'admin';
+    const keyVaultCollName = 'datakeys';
+    const keyVaultNamespace = `${keyVaultDbName}.${keyVaultCollName}`;
+    const kmsProviders = {
+      aws: {
+        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+        secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
+      },
+      local: {
+        key: Buffer.from(
+          'Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk',
+          'base64'
+        )
+      }
+    };
+
+    const noop = () => {};
+
+    let mongodbClientEncryption;
+    function getMongodbClientEncryption() {
+      mongodbClientEncryption =
+        mongodbClientEncryption || require('mongodb-client-encryption')(mongodb);
+      return mongodbClientEncryption;
+    }
+
+    describe('Data key and double encryption', function() {
+      // Data key and double encryption
+      // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      // First, perform the setup.
+      beforeEach(function() {
+        const mongodbClientEncryption = getMongodbClientEncryption();
+
+        // #. Create a MongoClient without encryption enabled (referred to as ``client``). Enable command monitoring to listen for command_started events.
+        this.client = this.configuration.newClient(
+          {},
+          { useNewUrlParser: true, useUnifiedTopology: true, monitorCommands: true }
+        );
+
+        this.commandStartedEvents = [];
+        this.client.on('commandStarted', e => {
+          if (!e.ismaster) {
+            this.commandStartedEvents.push(e);
+          }
+        });
+
+        const schemaMap = {
+          [dataNamespace]: {
+            bsonType: 'object',
+            properties: {
+              encrypted_placeholder: {
+                encrypt: {
+                  keyId: '/placeholder',
+                  bsonType: 'string',
+                  algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Random'
+                }
+              }
+            }
+          }
+        };
+
+        return (
+          Promise.resolve()
+            .then(() => this.client.connect())
+            // #. Using ``client``, drop the collections ``admin.datakeys`` and ``db.coll``.
+            .then(() => this.client.db(dataDbName).dropCollection(dataCollName))
+            .catch(noop)
```

Comment:
Are you just trying to ignore the `ns not found` error? Should we reuse [this function](https://github.com/mongodb/node-mongodb-native/blob/master/test/functional/apm_tests.js#L25-L27)

Reply:
yup